### ISSUE TITLE
Bug: Invalid autocorrection interaction between Lint/ParenthesesAsGroupedExpression & Style/RedundantParentheses

### ISSUE THREAD

**Issue by searls** [rubocop-hq/rubocop#8572]
In Standard @sethherr noticed an undesired interaction between the safe autocorrections of `Lint/ParenthesesAsGroupedExpression` and `Style/RedundantParentheses` (both enabled-by-default): https://github.com/testdouble/standard/issues/192

--------

## Expected behavior

Here's the starting code example:

```ruby
do_something (foo)
```

Here's what happens when I run only `Lint/ParenthesesAsGroupedExpression` and `Style/RedundantParentheses`:

```
$ rubocop --only "Lint/ParenthesesAsGroupedExpression,Style/RedundantParentheses" foo.rb 
Inspecting 1 file
W

Offenses:

foo.rb:1:13: W: Lint/ParenthesesAsGroupedExpression: (...) interpreted as grouped expression.
do_something (foo)
            ^
foo.rb:1:14: C: Style/RedundantParentheses: Don't use parentheses around a method call.
do_something (foo)
             ^^^^^

1 file inspected, 2 offenses detected
```

My expectation is the autocorrect would ultimately result in either `do_something foo` or `do_something(foo)`

## Actual behavior

When running the same command with autocorrect, this is what we get:

```
$ be rubocop --only "Lint/ParenthesesAsGroupedExpression,Style/RedundantParentheses" -a foo.rb
Inspecting 1 file
W

Offenses:

foo.rb:1:13: W: [Corrected] Lint/ParenthesesAsGroupedExpression: (...) interpreted as grouped expression.
do_something (foo)
            ^
foo.rb:1:14: C: [Corrected] Style/RedundantParentheses: Don't use parentheses around a method call.
do_something (foo)
             ^^^^^

1 file inspected, 2 offenses detected, 2 offenses corrected
```

And now the result is definitely wrong:

```
$ cat foo.rb 
do_somethingfoo

```

## Steps to reproduce the problem

I hope the instructions above are a clear enough minimal reproduction

## RuboCop version

```
$ rubocop -V
0.89.1 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.7.1 x86_64-darwin19)
```
